### PR TITLE
etcd config: handle decode err,  defaultkv option to avoid Read() err

### PR DESCRIPTION
**Pull request by keepstep**
1. ignore decode err in watch 
2. put /${prefix}/default "{}" to avoid Read() err

### CODE REVIEW

**Review by Davincible — CHANGES_REQUESTED** [review]

File: v4/config/source/etcd/etcd.go [hunk 1]
```diff
@@ -18,17 +18,29 @@ type etcd struct {
 	opts        source.Options
 	client      *clientv3.Client
 	cerr        error
+	// avoid err if no exist any keys with prefix
+	// put /prefix/default "{}"
+	defaultKv bool
```
[the remark this answers is not in the document]
Comment (keepstep) [reply v4/config/source/etcd/etcd.go]: have resolved

File: v4/config/source/etcd/util.go [hunk 2]
```diff
@@ -1,49 +1,62 @@
...
 
-	return data
+	return data, nil
 }
 
-func makeMap(e encoder.Encoder, kv []*mvccpb.KeyValue, stripPrefix string) map[string]interface{} {
+func makeMap(e encoder.Encoder, kv []*mvccpb.KeyValue, stripPrefix string) (map[string]interface{}, error) {
 	data := make(map[string]interface{})
-
+	var err error = nil
```
[the remark this answers is not in the document]
Comment (keepstep) [reply v4/config/source/etcd/util.go]: have resolved

File: v4/config/source/etcd/util.go [hunk 3]
```diff
@@ -1,49 +1,62 @@
...
 	clientv3 "go.etcd.io/etcd/client/v3"
 )
 
-func makeEvMap(e encoder.Encoder, data map[string]interface{}, kv []*clientv3.Event, stripPrefix string) map[string]interface{} {
+func makeEvMap(e encoder.Encoder, data map[string]interface{}, kv []*clientv3.Event, stripPrefix string) (map[string]interface{}, error) {
 	if data == nil {
 		data = make(map[string]interface{})
 	}
-
+	var err error = nil
```
[the remark this answers is not in the document]
Comment (keepstep) [reply v4/config/source/etcd/util.go]: have resolved

**Comment by keepstep** [comment]
Yes , I Have tested this also locally and make sure everything works correctly.

**Review by Davincible — APPROVED** [review]

**Pull request merged**